pkg/broker/receiver.go

```diff
@@ -179,24 +181,40 @@ func (r *Receiver) sendEvent(ctx context.Context, tctx cloudevents.HTTPTransport
 		return nil, err
 	}
 
+	// Set up the metrics context
+	ctx, _ = tag.New(ctx,
+		tag.Insert(TagTrigger, trigger.String()),
+		tag.Insert(TagBroker, fmt.Sprintf("%s/%s", trigger.Namespace, t.Spec.Broker)),
+	)
+	// Record event count and filtering time
+	startTS := time.Now()
+	defer func() {
+		dispatchTimeMS := int64(time.Now().Sub(startTS) / time.Millisecond)
+		stats.Record(ctx, MeasureTriggerDispatchTime.M(dispatchTimeMS))
+		stats.Record(ctx, MeasureTriggerEventsTotal.M(1))
+	}()
+
 	subscriberURIString := t.Status.SubscriberURI
 	if subscriberURIString == "" {
-		r.logger.Error("Unable to read subscriberURI")
+		ctx, _ = tag.New(ctx, tag.Upsert(TagResult, "error"))
 		return nil, errors.New("unable to read subscriberURI")
 	}
 	// We could just send the request to this URI regardless, but let's just check to see if it well
 	// formed first, that way we can generate better error message if it isn't.
 	subscriberURI, err := url.Parse(subscriberURIString)
 	if err != nil {
 		r.logger.Error("Unable to parse subscriberURI", zap.Error(err), zap.String("subscriberURIString", subscriberURIString))
+		ctx, _ = tag.New(ctx, tag.Upsert(TagResult, "error"))
 		return nil, err
 	}
 
-	if !r.shouldSendMessage(&t.Spec, event) {
+	if !r.shouldSendMessage(ctx, &t.Spec, event) {
 		r.logger.Debug("Message did not pass filter", zap.Any("triggerRef", trigger))
+		ctx, _ = tag.New(ctx, tag.Upsert(TagResult, "reject"))
```

Comment:
It seems like "ignore" or "filter" might be clearer here, as "reject" might indicate that the downstream sink returned a non-200 response code.

Reply:
Good point, what about `drop`? That's consistent with iptables definition (`REJECT` notifies the sender, `DROP` doesn't) https://serverfault.com/questions/157375/reject-vs-drop-when-using-iptables